Modules/VAMI/VAMI.psm1

```diff
@@ -621,53 +652,150 @@ Function New-VAMIUser {
     $createSpec.fullname = $fullname
     $createSpec.role = $role
     $createSpec.email = $email
     $createSpec.password = [VMware.VimAutomation.Cis.Core.Types.V1.Secret]$password
+    
+    if ((Get-CisService | Where-Object {$_.name -like "*accounts*"}).name -like "*techpreview*") {
+        $createSpec.username = $name
+        try {
+            Write-Host "Creating new user $name ..."
+            $userAPI.add($createSpec)
+        } catch {
+            Write-Error $Error[0].exception.Message
+        }
+    } else {
+        $createSpec.password_expires = $passwordexpires
+        $createSpec.password_expires_at = $passwordexpiresat
+        $createSpec.max_days_between_password_change = $maxpasswordage
+        try {
+            Write-Host "Creating new user $name ..."
+            $userAPI.create($name, $createSpec)
+        } catch {
+            Write-Error $Error[0].exception.Message
+        }
+    }
+}
 
-    try {
-        Write-Host "Creating new user $name ..."
-        $userAPI.add($createSpec)
-    } catch {
-        Write-Error $Error[0].exception.Message
+Function Update-VAMIUser {
+<#
+    .NOTES
+    ===========================================================================
+     Inspired by:    William Lam
+     Organization:  VMware
+     Blog:          www.virtuallyghetto.com
+     Twitter:       @lamw
+     Created by:    Michael Dunsdon
+     Twitter:      @MJDunsdon
+     Date:         September 16, 2020
+    ===========================================================================
+    .SYNOPSIS
+        This function to update fields of a VAMI local user using VAMI interface (5480)
+        for a VCSA node which can be an Embedded VCSA, External PSC or External VCSA.
+    .DESCRIPTION
+        Function to update fields of a VAMI local user
+    .EXAMPLE
+        Connect-CisServer -Server 192.168.1.51 -User administrator@vsphere.local -Password VMware1!
+        Update-VAMIUser -name lamw -fullname "William Lam" -role "operator" -email "lamw@virtuallyghetto.com" -password "VMware1!" -passwordexpires  -passwordexpiresat "1/1/1970" -maxpasswordage 90
+    .NOTES
+        Created script to allow updating of an exisiting user account. Script supports 6.5 and 6.7 VCSAs.
+#>
+    param(
+        [Parameter(Mandatory=$true)]
+        [String]$name,
+        [Parameter(Mandatory=$false)]
+        [String]$fullname,
+        [Parameter(Mandatory=$false)]
+        [ValidateSet("admin","operator","superAdmin")]
+        [String]$role,
+        [Parameter(Mandatory=$false)]
+        [String]$email="",
+        [Parameter(Mandatory=$false)]
+        [String]$password = $null,
+        [Parameter(Mandatory=$false)]
+        [switch]$passwordexpires,
+        [Parameter(Mandatory=$false)]
+        [String]$passwordexpiresat = $null,
+        [Parameter(Mandatory=$false)]
+        [String]$maxpasswordage = 90
+    )
+
+    if ((Get-CisService | Where-Object {$_.name -like "*accounts*"}).name -like "*techpreview*") {

```

Comment:
This Techpreview service check code can be extracted ins a separate function because it is used on multiple places in the code, e.g.

Test-ApplianceServiceIsTechPreview